### PR TITLE
modify the value of tunnel port

### PR DESCRIPTION
**Pull request by muxuelan**
**What type of PR is this?**
/kind bug

**What this PR does / why we need it**:
While tunnelport is found in "record.IPTunnelPort[localIP]"，it will equals nil without any err，causing panic that invalid memory address or nil pointer deference.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

**Review by fisherxu — COMMENTED** [review]
The cloudcore will panic when restart?

File: cloud/cmd/cloudcore/app/server.go [hunk 1]
```diff
@@ -155,9 +156,9 @@ func NegotiateTunnelPort() (*int, error) {
 			return nil, err
 		}
 
-		_, found = record.IPTunnelPort[localIP]
+		port, found = record.IPTunnelPort[localIP]
```
Comment (fisherxu) [inline cloud/cmd/cloudcore/app/server.go:159]:
```suggestion
		port, found = record.IPTunnelPort[localIP]
		if found {
		       return port,nil
		}
```

File: cloud/cmd/cloudcore/app/server.go [hunk 2]
```diff
@@ -155,9 +156,9 @@ func NegotiateTunnelPort() (*int, error) {
...
 		if !found {
```
Comment (fisherxu) [inline cloud/cmd/cloudcore/app/server.go:160]: remove the if here.

File: cloud/cmd/cloudcore/app/server.go [hunk 3]
```diff
@@ -206,11 +207,10 @@ func NegotiateTunnelPort() (*int, error) {
 			return nil, err
 		}
 
-		port := constants.ServerPort
-		return &port, nil
+		port = constants.ServerPort
 	}
 
-	return nil, nil
+	return &port, nil
```
Comment (fisherxu) [inline cloud/cmd/cloudcore/app/server.go:213]:
```suggestion
	return nil, error("failed to negotiate the tunnel port")
```

**Comment by muxuelan** [comment]
> The cloudcore will panic when restart?

Yes，tunnelport will equals nil directly which causes panic.

File: cloud/cmd/cloudcore/app/server.go (hunk 3 above)
Comment (muxuelan) [reply cloud/cmd/cloudcore/app/server.go:213]: Okay, I will

**Review by fisherxu — COMMENTED** [review]
/lgtm
/approve

**Comment by fisherxu** [comment]
@muxuelan Thanks for the fixing, could you please cherry-pick this patch to `release-1.7` branch(use hack/cherry_pick_pull.sh script)?

**Comment by kevin-wangzefeng** [comment]
Fixes: https://github.com/kubeedge/kubeedge/issues/2884

**Pull request merged**